### PR TITLE
add localization for template and fields via display name, expose breadcrumbs in types package

### PR DESCRIPTION
**Pull request by alexgomezlf**
- add optional displayName property for TemplateInfo and LfFieldInfo
- expose breadcrumbs types in types package

### CODE REVIEW

File: projects/ui-components/lf-metadata/field-components/field-base-parts/lf-field-base/base-field/base-field.directive.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { OnInit, Output, EventEmitter, Input, Directive, ViewChild, ElementRef, ChangeDetectorRef, OnDestroy } from '@angular/core';
+import { OnInit, Output, EventEmitter, Input, Directive, ViewChild, ElementRef, ChangeDetectorRef, OnDestroy, AfterViewInit } from '@angular/core';
```
Comment (alexgomezlf) [inline projects/ui-components/lf-metadata/field-components/field-base-parts/lf-field-base/base-field/base-field.directive.ts]: remove AfterViewInit

File: projects/ui-components/lf-metadata/field-components/lf-field-group/lf-field-group.component.html [hunk 2]
```diff
@@ -36,7 +36,7 @@
         <div *ngFor="let fieldDef of fieldDefinitions">
             <span class="lf-field-container" id="lf-field-container">
                 <span *ngIf="fieldDef?.fieldInfo.name" class="lf-field-title-container">
-                    <span class="lf-field-title">{{fieldDef?.fieldInfo.name}}</span>
+                    <span class="lf-field-title">{{fieldDef?.fieldInfo.displayName }}</span>
```
Comment (alexgomezlf) [inline projects/ui-components/lf-metadata/field-components/lf-field-group/lf-field-group.component.html]: add title

File: projects/ui-components/shared/lf-breadcrumbs/lf-breadcrumbs-public-api.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export * from './lf-breadcrumbs-types';
+export * from './lf-breadcrumbs.component';
+export * from './lf-breadcrumbs.module';
```
Comment (alexgomezlf) [inline projects/ui-components/shared/lf-breadcrumbs/lf-breadcrumbs-public-api.ts:3]: remove module?

File: package.json [hunk 4]
```diff
@@ -77,7 +77,7 @@
     "@angular/platform-browser": "^13.3.11",
     "@angular/platform-browser-dynamic": "^13.3.11",
     "@angular/router": "^13.3.11",
```
Comment (alexgomezlf) [inline package.json:79]: update changelog

File: projects/lf-documentation/src/app/lf-field-container-documentation/lf-field-container-demo.service.ts [hunk 5]
```diff
@@ -109,8 +121,8 @@ export class LfFieldContainerDemoService implements LfFieldContainerService {
 
   async getAvailableTemplatesAsync(): Promise<TemplateInfo[]> {
     const allTemplates: TemplateInfo[] = [
-      { id: 1, name: 'Lunch Orders' },
-      { id: 2, name: 'Email' }
+      { id: 1, name: 'Lunch Orders', displayName: 'Lunch Orders' },
```
Comment (Ke-Jin-Laserfiche) [inline projects/lf-documentation/src/app/lf-field-container-documentation/lf-field-container-demo.service.ts]: should this be the same as line 111?

File: projects/lf-documentation/src/app/lf-field-container-documentation/lf-field-container-demo.service.ts [hunk 6]
```diff
@@ -169,39 +187,45 @@ export class LfFieldContainerDemoService implements LfFieldContainerService {
             name: 'Subject',
             id: 11,
             fieldType: FieldType.String,
-            isRequired: true
+            isRequired: true,
+            displayName: "Subject"
```
Comment (Ke-Jin-Laserfiche) [inline projects/lf-documentation/src/app/lf-field-container-documentation/lf-field-container-demo.service.ts]: change this to single quote to be consistent

File: projects/lf-documentation/src/app/lf-field-documentation/lf-field-documentation.component.html [hunk 7]
```diff
@@ -62,4 +62,59 @@
...
+  <li>a: number</li>
+  <li>r: number</li>
+  <li>b: number</li>
+  <li>g: number</li>
+</ul>
+<p class="documentation-sub-header">
+  TemplateFieldInfo Interface
+</p>
+<ul>
+  <li>extends LfFieldInfo</li>
```
Comment (Ke-Jin-Laserfiche) [inline projects/lf-documentation/src/app/lf-field-documentation/lf-field-documentation.component.html]: just a thought, "extends LfFieldInfo" as bullet point reads a little weird. maybe move this one to the line above?
"TemplateFieldInfo Interface extends LfFieldInfo"

File: projects/ui-components/lf-metadata/lf-field-template-container/lf-field-template-container-demo.service.ts [hunk 8]
```diff
@@ -135,54 +147,63 @@ export const fieldInfosPerTemplate: { [templateId: number]: TemplateFieldInfo[]
...
       fieldType: FieldType.Date,
       rule: { ancestors: [DynamicFieldIds.ParentDate] },
+      displayName: 'dynamic child date'
     },
     {
       id: DynamicFieldIds.GrandparentTime,
       name: 'Grandparent Time',
       fieldType: FieldType.Time,
       rule: { ancestors: [] },
+      displayName: 'dynamic grandparenttime'
```
Comment (Ke-Jin-Laserfiche) [inline projects/ui-components/lf-metadata/lf-field-template-container/lf-field-template-container-demo.service.ts]: dynamic grandparenttime -> dynamic grandparent time

File: projects/ui-components/lf-metadata/lf-field-template-container/lf-field-template-container-demo.service.ts [hunk 9]
```diff
@@ -135,54 +147,63 @@ export const fieldInfosPerTemplate: { [templateId: number]: TemplateFieldInfo[]
...
       fieldType: FieldType.Time,
       rule: { ancestors: [DynamicFieldIds.ParentTime] },
+      displayName: 'dynamic child time'
     },
     {
       id: DynamicFieldIds.GrandparentDateTime,
       name: 'Grandparent DateTime',
       fieldType: FieldType.DateTime,
       rule: { ancestors: [] },
+      displayName: 'grandparent datetime'
```
Comment (Ke-Jin-Laserfiche) [inline projects/ui-components/lf-metadata/lf-field-template-container/lf-field-template-container-demo.service.ts]: why are these displayNames not started with "dynamic"?

Comment (alexgomezlf) [reply projects/ui-components/lf-metadata/lf-field-template-container/lf-field-template-container-demo.service.ts]: I missed those, updated

File: projects/lf-documentation/src/app/lf-field-container-documentation/lf-field-container-demo.service.ts (hunk 5 above)
Comment (alexgomezlf) [reply projects/lf-documentation/src/app/lf-field-container-documentation/lf-field-container-demo.service.ts]: add this as a const on the class

File: projects/lf-documentation/src/app/lf-field-container-documentation/lf-field-container-demo.service.ts (hunk 6 above)
Comment (alexgomezlf) [reply projects/lf-documentation/src/app/lf-field-container-documentation/lf-field-container-demo.service.ts]: this was updated in changelog commit

File: projects/lf-documentation/src/app/lf-field-documentation/lf-field-documentation.component.html (hunk 7 above)
Comment (alexgomezlf) [reply projects/lf-documentation/src/app/lf-field-documentation/lf-field-documentation.component.html]: moved the extends and changed bullet to read 'all properties of LfFieldInfo' but I can remove that if you think it's confusing

**Review by Ke-Jin-Laserfiche — APPROVED** [review]

**Pull request merged**